src/crawlee/_utils/requests.py

```diff
@@ -55,33 +56,21 @@ def normalize_url(url: str, *, keep_url_fragment: bool = False) -> str:
         A string containing the normalized URL.
     """
     # Parse the URL
-    parsed_url = urlparse(url.strip())
-    search_params = dict(parse_qsl(parsed_url.query))  # Convert query to a dict
+    parsed_url = URL(url.strip())
```

Comment:
Is the `strip` call necessary?

Reply:
Yep, `yarl` does not remove spaces at the end of a string